vaapi/vaapicontext.cpp

```diff
@@ -38,16 +91,13 @@ ConfigPtr VaapiConfig::create(const DisplayPtr& display,
         return ret;
     VAStatus vaStatus;
     VAConfigID config;
-    vaStatus = vaCreateConfig(display->getID(), profile, entry, attribList, numAttribs, &config);
 
-    // VAProfileH264Baseline is super profile for VAProfileH264ConstrainedBaseline
-    // old i965 driver incorrectly claims supporting VAProfileH264Baseline, but not VAProfileH264ConstrainedBaseline
-    if (vaStatus == VA_STATUS_ERROR_UNSUPPORTED_PROFILE
-        && profile == VAProfileH264ConstrainedBaseline
-        && entry == VAEntrypointVLD)
-        vaStatus = vaCreateConfig(display->getID(),
-                                  VAProfileH264Baseline,
-                                  VAEntrypointVLD, attribList, numAttribs, &config);
+    if (!checkProfileCompatible(display, profile)){
```

Comment:
for encoder we'd better think from other aspect. if they want encode a baseline profile. We give him a main profile. the target decoder may failed on encoded file.